src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedServiceTest.java

```diff
@@ -0,0 +1,158 @@
+
+        CaseDocument latestRefusalOrder = (CaseDocument)documentMap.get(CONTESTED_APPLICATION_NOT_APPROVED_LATEST_DOCUMENT);
+        assertThat(latestRefusalOrder.getDocumentUrl(),
+            is("http://document-management-store:8080/documents/015500ba-c524-4614-86e5-c569f82c718d"));
+        assertThat(latestRefusalOrder.getDocumentFilename(),
+            is("refusalOrderTestFilename.pdf"));
+        assertThat(latestRefusalOrder.getDocumentBinaryUrl(),
+            is("http://document-management-store:8080/documents/015500ba-c524-4614-86e5-c569f82c718d/binary"));
+    }
+
+    private CaseDetails contestedCaseDetails() throws Exception {
+        try (InputStream resourceAsStream = getClass().getResourceAsStream("/fixtures/refusal-order-contested.json")) {
+            return mapper.readValue(resourceAsStream, CallbackRequest.class).getCaseDetails();
+        }
+    }
+
+    private static void doCaseDocumentAssert(CaseDocument result) {
+        assertThat(result.getDocumentFilename(), is(FILE_NAME));
+        assertThat(result.getDocumentUrl(), is(DOC_URL));
+        assertThat(result.getDocumentBinaryUrl(), is(BINARY_URL));
+    }
+
+    private static class TestDocumentClient implements DocumentClient {
+
+        private Map<String, Object> value;
+
+        @Override
+        public Document generatePdf(DocumentGenerationRequest request, String authorizationToken) {
+            this.value = request.getValues();
+            return document();
+        }
+
+        @Override
+        public UUID bulkPrint(BulkPrintRequest bulkPrintRequest) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void deleteDocument(String fileUrl, String authorizationToken) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public DocumentValidationResponse checkUploadedFileType(String authorizationToken, String fileUrl) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Document stampDocument(Document document, String authorizationToken) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Document annexStampDocument(Document document, String authorizationToken) {
+            throw new UnsupportedOperationException();
+        }
+
+        void verifyAdditionalFieldsContested() {
+            Map<String, Object> data = data();
+
+            assertThat(data.get("ApplicantName"), is("Contested Applicant Name"));
+            assertThat(data.get("RespondentName"), is("Contested Respondent Name"));
+            assertThat(data.get("Court"),is("Nottingham County Court and Family Court"));
+            assertThat(data.get("JudgeDetails"), is("Her Honour Judge Contested"));
+            assertThat(data.get("ContestOrderNotApprovedRefusalReasonsFormatted"),
```

Comment:
can we get some tests for empty and one refusal reason also?

Reply:
Ill haggle you down on this one, edge cases are:
- no refusal reasons
- multiple refusal reasons
The test case for 1 refusal reason is a bit redundant, because if it works for multiple it works for one!
I can add a test case for empty refusal reasons